src/modV.js

```diff
@@ -258,6 +259,49 @@ class ModV extends EventEmitter2 {
 
 		return layerIndex;
 	}
+
+	/** @return {ModV.WorkersDataType} */
+	createWorkers() {
+		const palette = new PaletteWorker();
+		palette.on(PaletteWorker.EventType.PALETTE_ADDED, this.paletteAddHandler.bind(this));
+		palette.on(PaletteWorker.EventType.PALETTE_UPDATED, this.paletteUpdateHandler.bind(this));
+
+		return {
+			palette,
+		};
+	}
+
+	/**
+	 * @protected
+	 * @param {string} data
+	 */
+	paletteAddHandler(id) {
+		if(!this.palettes.has(id)) {
+			this.palettes.set(id, {});
+		} else {
+			console.error('Palette with ID', id, 'already exists');
+		}
+	}
+
+	/**
+	 * @protected
+	 * @param {string} id
+	 * @param {*} currentColor
+	 * @param {*} currentStep
+	 * @todo Types
+	 */
+	paletteUpdateHandler(id, currentColor, currentStep) {
+		this.palettes.set(id, {
+			currentColor: currentColor,
+			currentStep: currentStep
+		});
+	}
+
+	/** @param {string} id */
+	removePalette(id) {
```

Comment:
@2xAA About this method and above comments (similar methods).

I propose an event-driven approach. This will help to separate the application components and make them as independent as possible, to reduce coupling. The methods mentioned above is a reaction instance modV on external events, so they should be in class modV (and in paletteWorker we have strongly specific for worker code).

In this case, if we remove paletteWorker – the application will continue to work as if nothing had happened. Low coupling FTW.

If you disagree, then I'm glad to hear suggestion how to make it better.